ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/inmemory/InMemoryLedger.scala

```diff
@@ -170,6 +170,16 @@ class InMemoryLedger(
       acs.keys.get(key).filter(acs.isVisibleForStakeholders(_, forParty))
     })
 
+  override def lookupMaximumLedgerTime(contractIds: Set[AbsoluteContractId]): Future[Instant] =
+    Future.successful(this.synchronized {
```

Comment:
Wouldn't it be enough to synchronize on `acs` only?

Reply:
This class goes away soon and this just follows the same pattern used in the other methods.